src/main/java/com/dev/cinema/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,37 @@
+package com.dev.cinema.service.impl;
+
+import com.dev.cinema.exception.AuthenticationException;
+import com.dev.cinema.lib.Inject;
+import com.dev.cinema.lib.Service;
+import com.dev.cinema.model.User;
+import com.dev.cinema.service.AuthenticationService;
+import com.dev.cinema.service.UserService;
+import com.dev.cinema.util.HashUtil;
+import java.util.Optional;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> user = userService.findByEmail(email);
+        if (user.isEmpty()) {
+            throw new AuthenticationException("No User with such Email");
+        }
+        String hashedPassword = HashUtil.hashPassword(password, user.get().getSalt());
```

Comment:
let's refactor this method so that we would throw the same exception only once for the to `if` cases

Reply:
refactored